src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -40,21 +64,209 @@ public SnapshotBuilderGetConceptsResponse getConceptChildren(UUID datasetId, Int
+            (SnapshotBuilderProgramDataListCriteria) criteria;
+        return generateFilterForListCriteria(listCriteria);
+      }
+      case RANGE -> {
+        SnapshotBuilderProgramDataRangeCriteria rangeCriteria =
+            (SnapshotBuilderProgramDataRangeCriteria) criteria;
+        return generateFilterForRangeCriteria(rangeCriteria);
+      }
+      case DOMAIN -> {
+        SnapshotBuilderDomainCriteria domainCriteria = (SnapshotBuilderDomainCriteria) criteria;
+        return generateFilterForDomainCriteria(domainCriteria);
+      }
+      default -> throw new BadRequestException(
+          "Expected one of the range, list, or domain criteria");
+    }
+  }
+
+  @NotNull
+  private static BooleanAndOrFilterVariable generateFilterForRangeCriteria(
+      SnapshotBuilderProgramDataRangeCriteria rangeCriteria) {
+    TablePointer tablePointer = TablePointer.fromTableName("person");
+    TableVariable tableVariable = TableVariable.forPrimary(tablePointer);
+    return new BooleanAndOrFilterVariable(
+        BooleanAndOrFilterVariable.LogicalOperator.AND,
+        List.of(
+            new BinaryFilterVariable(
+                new FieldVariable(
+                    new FieldPointer(tablePointer, rangeCriteria.getName()), tableVariable),
+                BinaryFilterVariable.BinaryOperator.GREATER_THAN_OR_EQUAL,
+                new Literal(rangeCriteria.getLow().doubleValue())),
+            new BinaryFilterVariable(
+                new FieldVariable(
+                    new FieldPointer(tablePointer, rangeCriteria.getName()), tableVariable),
+                BinaryFilterVariable.BinaryOperator.LESS_THAN_OR_EQUAL,
+                new Literal(rangeCriteria.getHigh().doubleValue()))));
+  }
+
+  @NotNull
+  private static FunctionFilterVariable generateFilterForListCriteria(
+      SnapshotBuilderProgramDataListCriteria listCriteria) {
+    TablePointer tablePointer = TablePointer.fromTableName("person");
+    TableVariable tableVariable = TableVariable.forPrimary(tablePointer);
+    TablePointer conceptPointer = TablePointer.fromTableName("concept");
+    TableVariable conceptPointerVariable =
+        TableVariable.forJoined(
+            conceptPointer,
+            "concept_id",
+            new FieldVariable(
+                new FieldPointer(tablePointer, listCriteria.getName()), tableVariable));
+    return new FunctionFilterVariable(
+        FunctionFilterVariable.FunctionTemplate.IN,
+        new FieldVariable(new FieldPointer(conceptPointer, "concept_id"), conceptPointerVariable),
+        listCriteria.getValues().stream()
+            .map(value -> new Literal(value.doubleValue()))
+            .toArray(Literal[]::new));
+  }
+
+  private record OccurrenceTable(String tableName, String idColumnName) {}
+
+  private OccurrenceTable getOccurrenceTableFromDomainCriteria(
+      SnapshotBuilderDomainCriteria domainCriteria) {
+    // TODO: Load this information from snapshot builder settings, rather than hardcode
```

Comment:
👍 Great- I was hoping there would be a way to avoid some of the hardcoding in this code. Do we have a way to do this with this PR, or would this come later?